src/Data/Primitive/Vec.hs

```diff
@@ -259,6 +263,14 @@ packVec16 a b c d e f g h i j k l m n o p = runST $ do
   ByteArray ba# <- unsafeFreezeByteArray mba
   return $! Vec ba#
 
+replicateVecN :: forall a n . (KnownNat n, Prim a) => a -> Vec n a
```

Comment:
This doesn't really belong in this PR, but the function seems useful enough to keep around